### ISSUE TITLE
zPosition enhancement for eventViews

### ISSUE THREAD

**Issue by aditya-rastogi23** [richardtop/CalendarKit#262]
### New Issue Checklist

- [x] Use the latest version of CalendarKit
- [x] I've read the [Contribution Guidelines](https://github.com/richardtop/CalendarKit/blob/master/CONTRIBUTING.md)
- [x] I've searched for [existing GitHub issues](https://github.com/richardtop/CalendarKit/issues)

### Issue Description
I am trying to add zPosition to eventView.layer. Currently, there seems to be no right way to do it without editing the library code. (since I intend to use the library for long I would prefer not to edit library code). I will be willing to take up this issue and make a pull request. zPosition can be defaulted to 0. This issue occurs only when eventsWillOverlap is true. Essentially I have a hierarchy where for every day I have free time and working time and under them I have subEvents. However, if I call reloadData the events to move on top of each other in an undesired manner. By adding zPosition this issue will be resolved. Is there an enhancement you would be open to?

##### Code I'm using with CalendarKit

```Swift

[INSERT CODE HERE]

```

#### Result I am trying to achieve

[INSERT API OR UI MOCKUPS HERE]

**Comment by richardtop** [comment]
Hi, yes please go ahead with the pull request!
Also, could you please share some screenshots showing the issue and the desired behavior?

**Comment by aditya-rastogi23** [comment]
![Screenshot (8)](https://user-images.githubusercontent.com/31263366/85816657-99629a00-b789-11ea-954b-369d946e68ca.png)
![Screenshot (9)](https://user-images.githubusercontent.com/31263366/85816660-9a93c700-b789-11ea-8ae9-067c0a895ed9.png)

**Comment by richardtop** [comment]
@aditya-rastogi23 thanks for the idea, but I suggest changing the layout algorithm, as it could put smaller events on top of the larger ones and the beginning of the event on top of the end of the event.

**Comment by richardtop** [comment]
Could you please attach a project with the reproducible example of this issue?

**Comment by richardtop** [comment]
I'm not sure how you're getting the results as shown on the screenshot, as the current CalendarKit layout algorithm should produce the results as shown:

![Simulator Screen Shot - iPhone 11 Pro - 2020-07-13 at 15 36 04](https://user-images.githubusercontent.com/8013017/87305142-ac61c200-c51e-11ea-9431-8426d3a85497.png)


Please, provide a reproducible example or sample code you're using to achieve the result shown on the screenshot, or I'll have to close this issue.

**Comment by aditya-rastogi23** [comment]
Hey! I will be attaching the project in a couple of days. Having some issues with my setup. The screenshot you've provided I believe does not use the eventsWillOverlap as true, correct me if I am wrong.

**Comment by richardtop** [comment]
Thank you for getting back quickly! Yes, you're right. `eventsWillOverlap` feature is not commonly supported, so there might be issues.

Once you attach the project, I'll be able to help you troubleshoot the issue.

**Comment by aditya-rastogi23** [comment]
Hey! sorry for taking so long but you can find a sample project with this issue here:
https://github.com/aditya-rastogi23/calendarKitExample

**Comment by richardtop** [comment]
Hi, thanks for following up.

Few comments:

1. event reuse is actually disabled right now and could be completely removed from the library.
2. I'd also recommend taking a look at simpler solution: there is an option to set a z-position of the events happening later to a higher value than that of the events happening earlier.

E.g.

Event 1 - 10:00 - 13:00 - zPosition 0
Event 2 -12:00 - 14:00 - zPosition 1
.....


That would keep `zPosition` property out of the `Event` completely, so that the interface would be cleaner.

Later, I plan on introducing an option to plug in a custom layout algorithm, so that you'll be able to control and modify behaviors like that programmatically.

**Comment by aditya-rastogi23** [comment]
How am I supposed to access the zposition of events?(or access eventview in general) I don't want to alter the library code

**Comment by richardtop** [comment]
@aditya-rastogi23 indeed, I encourage you to alter the library code and submit a pull request which will be incorporated in the next version of the library. How does that sound?

**Comment by aditya-rastogi23** [comment]
Sounds good I'll start working on it

**Comment by sanath9** [comment]
HI aditya any update on that issue ?

**Comment by aditya-rastogi23** [comment]
My MacOS basically broke down, will not be able to work on it till august
